src/main/java/net/minecraftforge/fmp/client/multipart/IWeightedBakedModel.java

```diff
@@ -0,0 +1,104 @@
+package net.minecraftforge.fmp.client.multipart;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.block.model.ItemCameraTransforms;
+import net.minecraft.client.renderer.block.model.ItemOverrideList;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.util.EnumFacing;
+
+/**
+ * Interface that represents a model whose sides have different sorting priorities. Lower weights get added after higher
+ * weights, rendering on top of them and vice versa.
+ */
+public interface IWeightedBakedModel extends IBakedModel
+{
+    
+    /**
+     * Gets the weight of the quads on a specific side for the specified state.
+     */
+    public int getWeight(@Nullable IBlockState state, @Nullable EnumFacing side, long rand);
+
+    public static class WeightedModelAdapter implements IWeightedBakedModel
+    {
```

Comment:
The adapter's only purpose is to wrap a normal IBakedModel and allow you to externally provide weights without needeing to extend the class or create a completely new IWeightedBakedModel that forwards all the methods but one to the original model. People should only care about it if they have a model they want to depth-sort (in the case of Framez, I want to be able to depth-sort frames and covers so that the frame can either be on top or behind the cover).